lib/Basic/LangOptions.cpp

```diff
@@ -53,18 +64,20 @@ static const StringRef SupportedConditionalCompilationArches[] = {
   "wasm32",
 };
 
-static const StringRef SupportedConditionalCompilationEndianness[] = {
+static const SupportedConditionalValue SupportedConditionalCompilationEndianness[] = {
   "little",
   "big"
 };
 
-static const StringRef SupportedConditionalCompilationRuntimes[] = {
+static const SupportedConditionalValue SupportedConditionalCompilationRuntimes[] = {
   "_ObjC",
   "_Native",
 };
 
-static const StringRef SupportedConditionalCompilationTargetEnvironments[] = {
+static const SupportedConditionalValue SupportedConditionalCompilationTargetEnvironments[] = {
   "simulator",
+  { "macabi", "macCatalyst" },
+  "macCatalyst", // A synonym for "macabi" when compiling for iOS
```

Comment:
Since `targetEnvironment(macabi)` has been deprecated, we should probably consider `macabi` to be the synonym, not `macCatalyst`.

Reply:
I agree, but I'd like to do that in a separate PR.